src/vs/workbench/parts/extensions/electron-browser/extensionsViewlet.ts

```diff
@@ -436,24 +440,30 @@ export class ExtensionsViewlet extends ViewContainerViewlet implements IExtensio
 	}
 
 	private triggerSearch(immediate = false): void {
-		this.searchDelayer.trigger(() => this.doSearch(), immediate || !this.searchBox.getValue() ? 0 : 500).then(null, err => this.onError(err));
+		this.searchPromise.cancel();
+		const search = () => {
+			this.searchPromise = createCancelablePromise(token => this.doSearch(token));
+			return this.searchPromise;
+		};
+		this.searchDelayer.trigger(search, immediate || !this.searchBox.getValue() ? 0 : 500).then(null, err => this.onError(err));
+
 	}
 
 	private normalizedQuery(): string {
 		return this.searchBox.getValue().replace(/@category/g, 'category').replace(/@tag:/g, 'tag:').replace(/@ext:/g, 'ext:');
 	}
 
-	private doSearch(): Promise<any> {
+	private doSearch(token: CancellationToken): Promise<any> {
 		const value = this.normalizedQuery();
 		this.searchExtensionsContextKey.set(!!value);
 		this.searchBuiltInExtensionsContextKey.set(ExtensionsListView.isBuiltInExtensionsQuery(value));
 		this.groupByServersContextKey.set(ExtensionsListView.isGroupByServersExtensionsQuery(value));
 		this.recommendedExtensionsContextKey.set(ExtensionsListView.isRecommendedExtensionsQuery(value));
 		this.nonEmptyWorkspaceContextKey.set(this.contextService.getWorkbenchState() !== WorkbenchState.EMPTY);
 
-		if (value) {
+		if (value && !token.isCancellationRequested) {
 			return this.progress(Promise.all(this.panels.map(view => {
-				(<ExtensionsListView>view).show(this.normalizedQuery()).then(model => {
+				return (<ExtensionsListView>view).show(value, token).then(model => {
```

Comment:
When search is cancelled, we shouldnt be alerting the search result, as is done below.

I would suggest to update the `show` function to return `null` when search is cancelled.
Then add a null check on the model before running `alertSearchResult`

Reply:
`ExtensionsListView#show` now returns null if search was canceled